Documentation/api-reference.md

```diff
@@ -0,0 +1,380 @@
+```objectivec
++ (void) registerExtensions: (NSArray<Class*>* _Nonnull) extensions
+                 completion: (void (^ _Nullable)(void)) completion;
+```
+
+##### Example
+```objectivec
+@import AEPEdge;
+
+...
+[AEPMobileCore registerExtensions:@[AEPMobileEdge.class] completion:nil];
+```
+
+------
+
+### resetIdentities
+
+Resets current state of the AEP Edge extension and clears previously cached content related to current identity, if any.
+
+See [MobileCore.resetIdentities](https://aep-sdks.gitbook.io/docs/foundation-extensions/mobile-core/mobile-core-api-reference#resetidentities) for more details.
+
+------
+
+### sendEvent
+
+Sends an Experience event to the Adobe Experience Platform Edge Network
+
+#### Swift
+
+##### Syntax
+```swift
+static func sendEvent(experienceEvent: ExperienceEvent, _ completion: (([EdgeEventHandle]) -> Void)? = nil)
+```
+
+* _experienceEvent_ is the XDM [Experience Event](#experienceevent) sent to the Adobe Experience Platform Edge Network
+* _completion_ is an optional callback invoked when the request is complete and returns the associated [EdgeEventHandle](#edgeeventhandle)(s) received from the Adobe Experience Platform Edge Network. It may be invoked on a different thread.
+
+##### Example
+```swift
+//create experience event from dictionary:
+var xdmData : [String: Any] = ["eventType" : "SampleXDMEvent",
+                              "sample": "data"]
+let experienceEvent = ExperienceEvent(xdm: xdmData)
+```
+```swift
+// example 1 - send the experience event without handling the Edge Network response
+Edge.sendEvent(experienceEvent: experienceEvent)
+```
+```swift
+// example 2 - send the experience event and handle the Edge Network response onComplete
+Edge.sendEvent(experienceEvent: experienceEvent) { (handles: [EdgeEventHandle]) in
+            // handle the Edge Network response
+        }
+```
+
+#### Objective-C
+
+##### Syntax
+```objectivec
++ (void) sendExperienceEvent:(AEPExperienceEvent * _Nonnull) completion:^(NSArray<AEPEdgeEventHandle *> * _Nonnull)completion
+```
+
+##### Example
+```objectivec
+//create experience event from dictionary:
+NSDictionary *xdmData = @{ @"eventType" : @"SampleXDMEvent"};
+NSDictionary *data = @{ @"sample" : @"data"};
+```
+```objectivec
+// example 1 - send the experience event without handling the Edge Network response
+[AEPMobileEdge sendExperienceEvent:event completion:nil];
+```
+```objectivec
+// example 2 - send the experience event and handle the Edge Network response onComplete
+[AEPMobileEdge sendExperienceEvent:event completion:^(NSArray<AEPEdgeEventHandle *> * _Nonnull handles) {
+  // handle the Edge Network response
+}];
+```
+------
+
+### setLocationHint
+
+Set the Edge Network location hint used in requests to the Adobe Experience Platform Edge Network. Passing nil or an empty string will clear the existing location hint. Edge Network responses may overwrite the location hint to a new value when necessary to manage network traffic.
```

Comment:
- Set -> Sets the ..
- Passing nil or an empty string will clear the ... -> clears the existing..